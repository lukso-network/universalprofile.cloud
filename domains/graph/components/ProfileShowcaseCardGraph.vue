<script setup lang="ts">
type Props = {
  profile: Profile
}

const props = defineProps<Props>()
const profileBackground = useProfileBackground(props.profile, 300)
const profileAvatar = useProfileAvatar(props.profile, 80)
</script>

<template>
  <NuxtLink
    v-if="profile"
    :to="profileRoute(profile.address)"
    class="flex w-full"
  >
    <lukso-card
      variant="profile"
      :background-url="profileBackground?.url"
      :profile-url="profileAvatar?.url"
      shadow="small"
      is-hoverable
      custom-class="rounded-24"
      class="transition hover:scale-[1.01] active:scale-100"
      height="230"
    >
      <div slot="content" class="flex flex-col items-center px-4 pb-8 pt-1">
        <div class="flex">
          <lukso-username
            v-if="profile?.name"
            :name="profile?.name.toLowerCase()"
            :address="profile?.address"
            size="large"
            max-width="220"
          ></lukso-username>
          <lukso-username
            v-else
            name="anonymous-profile"
            :address="profile?.address"
            size="large"
            max-width="220"
            hide-prefix
          ></lukso-username>
        </div>
      </div>
    </lukso-card>
  </NuxtLink>
</template>
