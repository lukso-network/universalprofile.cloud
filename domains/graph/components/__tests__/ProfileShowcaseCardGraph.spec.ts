import { renderSuspended } from '@nuxt/test-utils/runtime'
import { beforeEach, describe, expect, it } from 'vitest'

import ProfileShowcaseCardGraph from '../ProfileShowcaseCardGraph.vue'

describe('ProfileShowcaseCardGraph', () => {
  beforeEach(() => {
    useIntl().setupIntl(defaultConfig)
  })

  it('should render empty state', async () => {
    const component = await renderSuspended(ProfileShowcaseCardGraph)

    expect(component.html()).toMatchSnapshot()
  })

  it('should render profile', async () => {
    const component = await renderSuspended(ProfileShowcaseCardGraph, {
      props: {
        profile: {
          address: '0x123',
          name: 'John Doe',
          backgroundImage: [
            {
              url: 'https://example.com/background-image.jpg',
              width: 100,
              height: 100,
            },
          ],
          profileImage: [
            {
              url: 'https://example.com/profile-image.jpg',
              width: 100,
              height: 100,
            },
          ],
        } as Profile,
      },
    })

    expect(component.html()).toMatchSnapshot()
  })
})
