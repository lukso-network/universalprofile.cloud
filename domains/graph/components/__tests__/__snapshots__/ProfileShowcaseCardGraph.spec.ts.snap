// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ProfileShowcaseCardGraph > should render empty state 1`] = `
"<div id="test-wrapper">
  <!--v-if-->
</div>"
`;

exports[`ProfileShowcaseCardGraph > should render profile 1`] = `
"<div id="test-wrapper"><a href="/0x123" class="flex w-full">
    <lukso-card variant="profile" background-url="https://example.com/background-image.jpg" profile-url="https://example.com/profile-image.jpg" shadow="small" is-hoverable="" custom-class="rounded-24" class="transition hover:scale-[1.01] active:scale-100" height="230">
      <div slot="content" class="flex flex-col items-center px-4 pb-8 pt-1">
        <div class="flex">
          <lukso-username name="john doe" address="0x123" size="large" max-width="220"></lukso-username>
        </div>
      </div>
    </lukso-card>
  </a></div>"
`;
